test/src/server/routes/entity/entity.js

```diff
@@ -80,3 +84,436 @@ describe('getDefaultAliasIndex', () => {
+		const relationshipTypeData = {
+			description: 'test descryption',
+			id: 1,
+			label: 'test label',
+			linkPhrase: 'test phrase',
+			reverseLinkPhrase: 'test reverse link phrase',
+			sourceEntityType: 'Author',
+			targetEntityType: 'Work'
+		};
+
+		relationshipTypeData.id = random.number();
+		await new RelationshipType(relationshipTypeData)
+			.save(null, {method: 'insert'});
+
+		// Relate Relationship Data with Relationship Type
+		relationshipDataA.typeId = relationshipTypeData.id;
+		relationshipDataA.id = random.number();
+
+		relationshipDataB.typeId = relationshipTypeData.id;
+		relationshipDataB.id = random.number();
+
+		// Create Relationships for Entities
+		const relationshipA = await new Relationship(relationshipDataA)
+			.save(null, {method: 'insert'});
+
+		const relationshipB = await new Relationship(relationshipDataB)
+			.save(null, {method: 'insert'});
+
+		// Create Relationship set ids for entities
+		const authorRelationshipSetId = random.number();
+		const workARelationshipSetId = random.number();
+		const workBRelationshipSetId = random.number();
+
+		// Create Relationship sets
+		let authorRelationshipSet = await new RelationshipSet({id: authorRelationshipSetId})
+			.save(null, {method: 'insert'})
+			.then(
+				(model) =>
+					model.relationships().attach([relationshipA, relationshipB]).then(() => model)
+			);
+		authorRelationshipSet = await authorRelationshipSet.refresh({withRelated: 'relationships'});
+
+		let workARelationshipSet = await new RelationshipSet({id: workARelationshipSetId})
+			.save(null, {method: 'insert'})
+			.then(
+				(model) =>
+					model.relationships().attach([relationshipA]).then(() => model)
+			);
+		workARelationshipSet = await workARelationshipSet.refresh({withRelated: 'relationships'});
+
+		let workBRelationshipSet = await new RelationshipSet({id: workBRelationshipSetId})
+			.save(null, {method: 'insert'})
+			.then(
+				(model) =>
+					model.relationships().attach([relationshipB]).then(() => model)
+			);
+		workBRelationshipSet = await workBRelationshipSet.refresh({withRelated: 'relationships'});
+
+		// Set the Entity Attributes as Required
+		author.set('revisionId', revisionId);
+		author.set('relationshipSetId', authorRelationshipSet.get('id'));
+		workA.set('revisionId', revisionId);
+		workA.set('relationshipSetId', workARelationshipSet.get('id'));
+		workB.set('revisionId', revisionId);
+		workB.set('relationshipSetId', workBRelationshipSet.get('id'));
+
+		// Save the Entities and Update the database
+		await author.save(null, {method: 'update'});
+		await workA.save(null, {method: 'update'});
+		await workB.save(null, {method: 'update'});
+
+		// Get updated Entity
+		workA = await new Work({bbid: workA.get('bbid')}).fetch({withRelated: 'relationshipSet.relationships'});
+
+		// Get list of Affected Entities
+		const affectedEntities = await deleteRelationships(orm, null, workA.toJSON());
+
+		expect(affectedEntities).to.have.length(1);
```

Comment:
Here let's also ensure it's the right affected entity bbid  you end up with (the author)

You also want to make sure the author's new relationship set 1. has a different id than before and 2. that it still contains a relationship to workB

Reply:
There's one item that you did not address:
You also want to make sure the author's new relationship set has a different id than before:
authorRelationshipSetId should not be equal to affectedEntityRelationshipSet.get('id')